go/tools/builders/compilepkg.go

```diff
@@ -273,7 +273,9 @@ func compileArchive(
 		hSrcs[i] = src.filename
 	}
 	haveCgo := len(cgoSrcs)+len(cSrcs)+len(cxxSrcs)+len(objcSrcs)+len(objcxxSrcs) > 0
-	packageUsesCgo := cgoEnabled && haveCgo
+	// NOTE: We build cgo packages with cgoEnabled both set to true and false.
```

Comment:
I have a hard time fully understanding this comment, could you expand it?

Reply:
Great point. This is unclear to me now, a week later :)

I revised the comments on both variables to try to make this more clear. Basically the problem with assembly specifically is:

* if the package contains any cgo files, the assembler is the C compiler.
* Any package, including cgo packages, can be built with CGO_ENABLED=0 AND with CGO_ENABLED=1. When there are build constraints, this should work.

These two things are related but independent choices. Previously: disabling CGO would cause this code to revert to using Go's assembler for the .s files.